src/reformat_libyuv.c

```diff
@@ -77,6 +77,54 @@ avifResult avifImageRGBToYUVLibYUV(avifImage * image, const avifRGBImage * rgb)
     return AVIF_RESULT_NOT_IMPLEMENTED;
 }
 
+// Two-step replacement for AVIF_RGB_FORMAT_RGBA to 8-bit BT.601 full range YUV, which is missing from libyuv.
+static int avifABGRToJ420(const uint8_t * src_abgr,
+                          int src_stride_abgr,
+                          uint8_t * dst_y,
+                          int dst_stride_y,
+                          uint8_t * dst_u,
+                          int dst_stride_u,
+                          uint8_t * dst_v,
+                          int dst_stride_v,
+                          int width,
+                          int height)
+{
+    // A temporary buffer is needed to swap the R and B channels before calling ARGBToJ420().
+    uint8_t * src_argb;
+    const int src_stride_argb = width * 4;
+    const uint64_t soft_allocation_limit = 16384; // Arbitrarily chosen trade-off between CPU and memory footprints.
+    int num_allocated_rows;
+    if ((height == 1) || ((uint64_t)src_stride_argb * height <= soft_allocation_limit)) {
+        // Process the whole buffer in one go.
+        num_allocated_rows = height;
+    } else {
+        // The last row of an odd number of RGB rows to be converted to subsampled YUV is treated differently
+        // by libyuv, so make sure all steps but the last one process an even number of rows.
+        // Try to process as many row pairs as possible in a single step without allocating more than
+        // soft_allocation_limit, unless two rows need more than that.
+        num_allocated_rows = AVIF_MAX(1, soft_allocation_limit / ((uint64_t)src_stride_argb * 2)) * 2;
+    }
+    src_argb = avifAlloc((uint64_t)num_allocated_rows * src_stride_argb);
```

Comment:
Cast to `size_t` instead (see lines 119-122), because `uint64_t` is too big for `avifAlloc()` in 32-bit builds.

Note: This cast seems unnecessary, because this product is at most `soft_allocation_limit` or `2 * src_stride_argb`. We already know `src_stride_argb <= src_stride_abgr <= INT_MAX`. We just need to make sure `2 * src_stride_argb <= INT_MAX`.